### ISSUE TITLE
MS_EXPOSE_BUF misses internal state exposure when returning via `ByteBuffer.array()`

### ISSUE THREAD

**Issue by 9iang22** [spotbugs/spotbugs#4007]
### Summary
`FindReturnRef` detects many `EI/MS_EXPOSE_*` patterns (direct getter/setter exposure, `Buffer.duplicate()`, `Buffer.wrap()`), but appears to miss two related exposure cases under the same rule intent:
1. Returning backing arrays via `ByteBuffer.array()`
2. Returning via `Arrays.copyOf(...)` for arrays of mutable elements (e.g., `Date[]`), where the copy is only shallow

### Code to Reproduce
```java
import java.nio.ByteBuffer;
import java.util.Arrays;
import java.util.Date;
public class FindReturnRefCombinedRepro {
  private final ByteBuffer buf = ByteBuffer.allocate(16);
  private static final ByteBuffer S_BUF = ByteBuffer.allocate(16);
  private Date[] dates = new Date[] { new Date() };
  // Case 1A: instance backing array exposure (missed)
  public byte[] getArray() {
    return buf.array();
  }
  // Case 1B: static backing array exposure (missed)
  public static byte[] getStaticArray() {
    return S_BUF.array();
  }
  // Detected: related pattern that is detected
  public ByteBuffer getDuplicate() {
    return buf.duplicate();  // detected
  }
  // Case 2: shallow copy of mutable element array (missed)
  public void setDates(Date[] input) {
    dates = Arrays.copyOf(input, input.length);
  }
  // Detected: clone-return pattern that is detected
  public Date[] getDatesClone() {
    return dates.clone(); // detected
  }
}
```
Run the following command
```
javac -d classes FindReturnRefCombinedRepro.java
spotbugs -textui -effort:max -low classes
```

### Actual Result
- Reported: getDuplicate() (buffer duplicate exposure)
- Reported: getDatesClone() (array clone exposure)
- Not reported: getArray() / getStaticArray() (ByteBuffer.array())
- Not reported: setDates(...) using Arrays.copyOf(...) on Date[]
### Expected Result
- getArray() / getStaticArray() should be flagged as EI/MS internal representation exposure because array() returns a mutable backing array alias.
- setDates(Date[]) with Arrays.copyOf(...) should be flagged (or documented as unsupported) for mutable element types, since copying is shallow and aliases Date objects.

### Why this matters
Both patterns allow external code to mutate state that the object intends to encapsulate:
- ByteBuffer.array() exposes internal storage directly.
- Arrays.copyOf(Date[]) creates a new array, but still shares mutable element references.

### Version
- SpotBugs 4.9.8
- JDK 21

**Comment by 9iang22** [comment]
Close as fixed in #4235
